pkg/hooks/hooks.go

```diff
@@ -0,0 +1,70 @@
+package hooks
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/rancher/terraform-controller/pkg/provider"
+	"github.com/rancher/terraform-controller/pkg/provider/github"
+	"github.com/rancher/terraform-controller/pkg/types"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+type WebhookHandler struct {
+	providers []provider.Provider
+}
+
+func newHandler(rContext *types.Context) *WebhookHandler {
+	wh := &WebhookHandler{
+		providers: []provider.Provider{
+			// register all supported webhook handler here
+			github.NewGitHub(rContext),
+		},
+	}
+
+	return wh
+}
+
+func (h *WebhookHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	code, err := h.execute(req)
+	if err != nil {
+		e := map[string]interface{}{
+			"type":    "error",
+			"code":    code,
```

Comment:
in case of map[string]string , convert code
```suggestion
"code":    strconv.Itoa(code),
```